### PR TITLE
[ refactor ] Remove Data.Strings module

### PR DESCRIPTION
**Pull request by ska80**
This PR ~~is a follow-up to #1606 and~~ removes `Data.Strings` completely.

~~Do not merge until the next version (0.5.0) is released.~~

Also, I had to add Markdown Linter rules to please the linter.

### CODE REVIEW

**Comment by ska80** [comment]
Will reopen when it is ok to merge.

**Comment by gallais** [comment]
Leave it open but as a draft?

**Comment by fabianhjr** [comment]
Hey @ska80 I think we don't have to wait until 0.5.0 but rather update the CI from 0.3.0 to 0.4.0

**Comment by ska80** [comment]
> Hey @ska80 I think we don't have to wait until 0.5.0 but rather update the CI from 0.3.0 to 0.4.0 (Eg #1601)

At first, I wanted to submit this PR as a single patch but building the Idris compiler from the previous version failed due to the absence of `Data.String` in 0.3.0.  Hence I had to split it in two phases.

**Comment by ska80** [comment]
Reopening and marking it as a draft.

**Comment by fabianhjr** [comment]
You could also change this line from 0.3.0 to 0.4.0: https://github.com/idris-lang/Idris2/blob/ea1ad1688f1e8ff275b229c733d691d0b1dbe107/.github/workflows/ci-ubuntu-combined.yml#L15

**Comment by fabianhjr** [comment]
Ah wait I forgot that `bootstrap-build` was renamed `bootstrap` let me check all the changes.

**Comment by fabianhjr** [comment]
https://github.com/idris-lang/Idris2/blob/ea1ad1688f1e8ff275b229c733d691d0b1dbe107/.github/workflows/ci-ubuntu-combined.yml#L89

This from `bootstrap-build` to `bootstrap`

**Comment by fabianhjr** [comment]
Well, there it goes, it passes :3

**Comment by ska80** [comment]
> Well, there it goes, it passes :3

So it could be merged directly then? @gallais 

Should I close #1606 ?

**Comment by fabianhjr** [comment]
I would prefer to get #1601 merged first since that is specifically about bringing the CI up to 0.4.0 and speeds it up a lot.

File: .github/workflows/ci-ubuntu-combined.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ jobs:
           wget https://www.idris-lang.org/idris2-src/idris2-$IDRIS2_VERSION.tgz
           tar zxvf idris2-$IDRIS2_VERSION.tgz
           cd Idris2-$IDRIS2_VERSION
-          make bootstrap-build
+          make bootstrap
```
Comment (gallais) [inline .github/workflows/ci-ubuntu-combined.yml:89]: Why this unrelated change?

Comment (ska80) [reply .github/workflows/ci-ubuntu-combined.yml:89]: > Why this unrelated change?

This change was suggested by @fabianhjr 

https://github.com/idris-lang/Idris2/pull/1607#issuecomment-867676641

Comment (gallais) [reply .github/workflows/ci-ubuntu-combined.yml:89]: So if I'm reading this correctly: https://github.com/idris-lang/Idris2/commit/b81b390f20f17f6beddf0c0a14f2463eafe0ceb0
all of this time we've been running the wrong command & it would not actually test anything?

Comment (fabianhjr) [reply .github/workflows/ci-ubuntu-combined.yml:89]: bootstrap-build was renamed bootstrap between v0.3.0 and v0.4.0
